### DISCUSSION TITLE
PeppolDirectory not found

### DISCUSSION THREAD

**Discussion by fatman1221** [phax/phoss-smp#250] [General]
![image](https://github.com/phax/phoss-smp/assets/73154573/56a0204a-38be-46ab-bca8-7d4dcf9f7e58)

Hi,
The log shows that the call was successful, but it cannot be found in the peppol directory.
https://test-directory.peppol.eu/

**Comment by fatman1221** [comment]
![image](https://github.com/phax/phoss-smp/assets/73154573/f6b71fac-dbf1-465a-b16d-95fb9efcf546)
![image](https://github.com/phax/phoss-smp/assets/73154573/b47b54d2-0d33-4db4-ba1c-4ed54cd091f2)

**Comment by phax** [comment]
The problem is, that your SMP is not accessible correctly from the outside world.
For this particular participant, the outside access URL is http://B-36fae5e51375f6b8c57a567953c58081.iso6523-actorid-upis.acc.edelivery.tech.ec.europa.eu 

When opening this URL in the browser, I am getting redirected to https://dnspod.qcloud.com/static/webblock.html?....

So please resolve this issue and open up your SMP to the world - then the Directory indexation will also work

**Reply by fatman1221** [reply]
Thank you for your reply. 
I have changed the IP address and the above issue has been resolved. 
However, there are two questions: 
1. Can I add a context address for this? 
3. Must it be HTTP? Does it support HTTPS

**Reply by fatman1221** [reply]
![image](https://github.com/phax/phoss-smp/assets/73154573/6766cc19-83a0-4f25-b5c1-14fcf1672f30)
Can I achieve this effect?

**Reply by phax** [reply]
Super you progressed.
* Unfortunately, the context path cannot be used. You must run on root path ("/"). Use the property `smp.forceroot=true` to remove it from the outside view
* And yes, it must be HTTP. See chapter 5.1 of the Peppol SMP specification. This is necessary due to the way, how the SMP is accessed with the dynamic DNS names.

Btw. I still can't access http://b-36fae5e51375f6b8c57a567953c58081.iso6523-actorid-upis.acc.edelivery.tech.ec.europa.eu/ from the Internet

**Reply by fatman1221** [reply]
I registered using IP 124.222.38.115 for the first time, but cannot access it through dns name b-36fae5e51375f6b8c57a567953c58081iso6523 vector upis. acc. delivery. tech. ec. europa. eu, Later, I changed the IP address to 18.234.238.21 and now I am using 18.234.238.21. The DNS name B-2402a303befcba74f7541353399249cf.iso6523 vector upis. acc. delivery. tech. ec. europa. eu for this IP address is automatically redirected to 124.222.38.115

**Reply by fatman1221** [reply]
![image](https://github.com/phax/phoss-smp/assets/73154573/06d4e2c5-6631-4905-af9b-dae14810f5ba)
This is something I don't quite understand, why did I jump to http://124.222.38.115/public

I can get the data through this

http://b-2402a303befcba74f7541353399249cf.iso6523-actorid-upis.acc.edelivery.tech.ec.europa.eu/iso6523-actorid-upis%3A%3A9919%3Aeasipass/services/busdox-docid-qns%3A%3Aurn%3Aoasis%3Anames%3Aspecification%3Aubl%3Aschema%3Axsd%3AInvoice-2%3A%3AInvoice%23%23urn%3Acen.eu%3Aen16931%3A2017%23compliant%23urn%3Afdc%3Apeppol.eu%3A2017%3Apoacc%3Abilling%3A3.0%3A%3A2.1

I need your help. Thank you very much

**Reply by phax** [reply]
So okay, you registered your SMP to the SML - good. But how did you do it?

In the below image, it is recommended to use "1.1.1.1" as the physical address, because this parameter is not used. Instead use the "logical address" and provide a real domain name and not just an IP address.

![grafik](https://github.com/phax/phoss-smp/assets/196219/164e61a7-0e33-4c18-b1ef-9f609500e09c)

So e.g. for my test SMP at smp.helger.com I used "1.1.1.1" as the physical address and "http://smp.helger.com" as the logical address and it works just fine. The difference is, if the SML creates an `A` record or a `CNAME` record for your SMP

Btw. I can't reach `b-2402a303befcba74f7541353399249cf.iso6523-actorid-upis.acc.edelivery.tech.ec.europa.eu` via `nslookup` so did you perform the above registration process??? Do you have a public domain for your IP address?

Please check your setup against the Peppol specific documentation at https://github.com/phax/phoss-smp/blob/master/docs/Setting%20up%20an%20SMP%20for%20Peppol.pdf

**Reply by fatman1221** [reply]
you can click  this link 
http://18.234.238.21/secure/locale-en_US/menuitem-service_groups
I initially registered using a machine with the IP address 124.222.38.115. However, I later changed my IP address to 18.234.238.21 while using the same WAR package. Now, after entering my username and password, I am automatically redirected to a different page. The IP address of this page, 124.222.38.115, is the one I used during my first registration. If I go back one step, I can access the system. I'm a bit confused by this. If you could log in and reproduce this phenomenon, I would greatly appreciate it.

**Reply by fatman1221** [reply]
I've identified the issue. I had set the configuration in the properties file to "smp.publicurl = http://124.222.38.115/". Thank you very much for your assistance; everything is working fine now.

**Reply by phax** [reply]
Yes, I can reach it :)

Your `smp.publicurl` value in the `application.properties` is still set to `http://124.222.38.115/"` - fix this to get the redirect done
